### PR TITLE
Add functionality: convert object implementing interface Packet to bytes and from bytes to object implementing interface Packet

### PR DESCRIPTION
**Pull request by ghost**
add extending from Serializable to interface Packet
interface PacketConverter - 2 methods toBytes(), toPacket()
implementation of PacketConverter ->  NetworkPacketConverter
add test for PacketFactory, NetworkPacketConverter

### CODE REVIEW

File: common/src/test/java/com/blackship/battlesheep/common/communication/network/packet/PacketFactoryTest.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    @Test
+    public void shouldCreateNetworkMovePacket() {
+
+        //given
+
+        //when
+        Packet givenPacket = PacketFactory.createMove();
+
+        //then
+        assertTrue(givenPacket instanceof NetworkPacketMove);
```
Comment (arturAR) [inline common/src/test/java/com/blackship/battlesheep/common/communication/network/packet/PacketFactoryTest.java:23]: Evviva l'arte!

File: common/src/main/java/com/blackship/battlesheep/common/communication/packet/Packet.java [hunk 2]
```diff
@@ -2,9 +2,10 @@
...
 import com.blackship.battlesheep.common.communication.packet.enums.PacketType;
 
+import java.io.Serializable;
 import java.time.LocalTime;
 
-public interface Packet {
+public interface Packet extends Serializable {
     PacketType getPacketType();
     LocalTime getCreationTime();
     Packet setCreationTime(LocalTime time);
```
Comment (arturAR) [inline common/src/main/java/com/blackship/battlesheep/common/communication/packet/Packet.java:11]: Do you need this? YAGNI?

File: common/src/main/java/com/blackship/battlesheep/common/communication/network/packet/NetworkPacketConverter.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    public byte[] toByte(Packet packet) throws IOException {
+        ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+        new ObjectOutputStream(byteArrayOutputStream).writeObject(packet);
+
+        return byteArrayOutputStream.toByteArray();
+    }
+
+    @Override
+    public Packet toPacket(byte[] bytes) throws IOException, ClassNotFoundException {
+        return (Packet) new ObjectInputStream(new ByteArrayInputStream(bytes)).readObject();
```
Comment (arturAR) [inline common/src/main/java/com/blackship/battlesheep/common/communication/network/packet/NetworkPacketConverter.java:24]: Hard to read, try to divide to few lines.

**Review by arturAR — APPROVED** [review]

File: common/src/main/java/com/blackship/battlesheep/common/communication/packet/Packet.java (hunk 2 above)
Comment (ghost) [reply common/src/main/java/com/blackship/battlesheep/common/communication/packet/Packet.java:11]: yes

File: common/src/main/java/com/blackship/battlesheep/common/communication/network/packet/NetworkPacketConverter.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
+package com.blackship.battlesheep.common.communication.network.packet;
+
+import com.blackship.battlesheep.common.communication.packet.Packet;
+import com.blackship.battlesheep.common.communication.packet.PacketConverter;
+
+import java.io.*;
```
Comment (piotr921) [inline common/src/main/java/com/blackship/battlesheep/common/communication/network/packet/NetworkPacketConverter.java:6]: Import of '*'

File: common/src/main/java/com/blackship/battlesheep/common/communication/packet/Packet.java (hunk 2 above)
Comment (piotr921) [reply common/src/main/java/com/blackship/battlesheep/common/communication/packet/Packet.java:11]: :pig_nose:

File: common/src/test/java/com/blackship/battlesheep/common/communication/network/packet/NetworkPacketConverterTest.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
+package com.blackship.battlesheep.common.communication.network.packet;
+
+import com.blackship.battlesheep.common.communication.packet.Packet;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+
+import static org.testng.Assert.*;
```
Comment (piotr921) [inline common/src/test/java/com/blackship/battlesheep/common/communication/network/packet/NetworkPacketConverterTest.java:9]: import '*'

File: common/src/test/java/com/blackship/battlesheep/common/communication/network/packet/PacketFactoryTest.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
+package com.blackship.battlesheep.common.communication.network.packet;
+
+import com.blackship.battlesheep.common.communication.packet.Packet;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.*;
```
Comment (piotr921) [inline common/src/test/java/com/blackship/battlesheep/common/communication/network/packet/PacketFactoryTest.java:6]: don't import '*'

File: common/src/test/java/com/blackship/battlesheep/common/communication/network/packet/PacketFactoryTest.java (hunk 1 above)
Comment (piotr921) [reply common/src/test/java/com/blackship/battlesheep/common/communication/network/packet/PacketFactoryTest.java:23]: it's always true according to intellij, and it was classified as "Probable bug" :exclamation:

**Review by piotr921 — APPROVED** [review]

**Pull request merged**